backend/app/controllers/MenteesController.php

```diff
@@ -38,6 +42,101 @@ public function singleMentee($id) {
         ]);
     }
 
+    public function login()
+    {
+
+        $credentials = request()->get(['email', 'password']);
+
+        if(!$credentials['email'] || !$credentials['password']) {
+            response()->exit([
+                'error' => '404',
+                'message' => "One of required fields isn't filled.",
+            ]);
+        }
+        $user = User::where('email', $credentials['email'])->first();
+
+        $user = auth()->login($credentials);
+
+        if (!$user) {
+            response()->exit(auth()->errors());
+        }
+
+        response()->json($user);
+    }
+
+    public function menteeInfo()
+    {
+        if (auth()->user()) {
+            $id = auth()->id();
+
```

Comment:
We should have a prevalidation middleware and use it in protected routes. Check this:
https://leafphp.dev/modules/auth/protecting-your-routes.html#using-middleware

Reply:
This way every time a request reaches a protected route we don't have to write the same check (if user is logged in)

Reply:
This might be also helpful as we're using controllers:
https://leafphp.dev/modules/router/middleware.html#%E2%8F%B3-before-route-middlewares

Reply:
Brief explanation: https://leafphp.dev/docs/routing/middleware.html#middleware-route-option